Meta/run.py

```diff
@@ -0,0 +1,908 @@
+                    "i82801b11-bridge,bus=bridge1,id=bridge2",
+                    "sdhci-pci,bus=bridge2",
+                    "i82801b11-bridge,id=bridge3",
+                    "sdhci-pci,bus=bridge3",
+                    "ich9-ahci,bus=bridge3",
+                ]
+            )
+            config.character_devices.append("stdio,id=stdout,mux=on")
+            config.enable_usb = True
+
+    # Modifications for machine types that are *mostly* like the default,
+    # but not entirely (especially in terms of networking).
+    match config.machine_type:
+        case MachineType.QEMUWithoutNetwork | MachineType.QEMU35Grub:
+            config.network_backend = None
+            config.network_default_device = config.ethernet_device_type
+            config.packet_logging_arguments = []
+        case MachineType.QEMUTap:
+            config.network_backend = "tap,ifname=tap0,id=br0"
+            config.network_default_device = f"{config.ethernet_device_type},netdev=br0"
+        case MachineType.QEMUGrub | MachineType.QEMUExtLinux:
+            config.kernel_cmdline = []
+
+    if config.qemu_kind != QEMUKind.NativeWindows:
+        config.extra_arguments.extend(["-qmp", "unix:qmp-sock,server,nowait"])
+
+    config.extra_arguments.extend(["-name", "SerenityOS", "-d", "guest_errors"])
+
+
+def assemble_arguments(config: Configuration) -> list[str | Path]:
+    if config.machine_type == MachineType.Bochs:
+        boch_src = Path(config.serenity_src or ".", "Meta/bochsrc")
+        return [config.bochs_binary, "-q", "-f", boch_src]
+
+    return [
+        config.qemu_binary or "",
+        # Deviate from standard order here:
+        # The device list contains PCI bridges which must be available for other devices.
+        *config.device_arguments,
+        *config.kernel_and_initrd_arguments,
+        *config.packet_logging_arguments,
+        *config.spice_arguments,
+        *config.extra_arguments,
+        *config.accelerator_arguments,
+        *config.kernel_cmdline_arguments,
+        *config.ram_arguments,
+        *config.cpu_arguments,
+        *config.smp_arguments,
+        *config.machine_arguments,
+        *config.usb_arguments,
+        *config.audio_backend_arguments,
+        *config.audio_devices_arguments,
+        *config.vga_arguments,
+        *config.display_backend_arguments,
+        *config.display_device_arguments,
+        *config.network_backend_arguments,
+        *config.network_default_arguments,
+        *config.boot_drive_arguments,
+        *config.character_device_arguments,
+    ]
+
+
+class TapController:
+    """Context manager for setting up and tearing down a tap device when QEMU is run with tap networking."""
+
+    def __init__(self, machine_type: MachineType):
+        self.should_enable_tap = machine_type == MachineType.QEMUTap
+
+    def __enter__(self) -> None:
+        if self.should_enable_tap:
+            run(["sudo", "ip", "tuntap", "del", "dev", "tap0", "mode", "tap"])
```

Comment:
I'm not sure if we should just go around and delete peoples tap devices if we don't know it's ours (at least to a certain degree of confidence). We should be able to pick any name we want for this, so we should probably think of something that isn't the default tap interface name.

Reply:
Eh, looks like the current script already does this, which makes it a thing that we might as well fix in the future. Marking as resolved.